pkg/kubelet/kuberuntime/logs/logs.go

```diff
@@ -236,10 +283,15 @@ func newLogWriter(stdout io.Writer, stderr io.Writer, opts *LogOptions) *logWrit
 
 // writeLogs writes logs into stdout, stderr.
 func (w *logWriter) write(msg *logMessage) error {
-	if msg.timestamp.Before(w.opts.since) {
-		// Skip the line because it's older than since
+	switch {
```

Comment:
This method is already huge, and this change makes it larger.  I'd like to see this split up into two paths (if one is substantially different), with a commit moving the existing path unchanged to its own method happening separately (so we can review that independently).  This is already hard to read, complex code, and then new code is also hard to read and complex, so we need to be reducing the difficulty of review and longer term support by focusing on clarity and clear separation.

Reply:
Sure! I would try to make the function easier to review